python/ray/serve/_private/replica_result.py

```diff
@@ -120,44 +96,40 @@ async def async_wrapper(self, *args, **kwargs):
     @_process_response
     def get(self, timeout_s: Optional[float]):
         assert (
-            self._obj_ref is not None or not self._is_streaming
-        ), "get() can only be called on a non-streaming ActorReplicaResult"
+            not self._is_streaming
+        ), "get() can only be called on a unary ActorReplicaResult."
 
         start_time_s = time.time()
-        self.resolve_gen_to_ref_if_necessary_sync(timeout_s)
-
+        object_ref = self.to_object_ref(timeout_s=timeout_s)
         remaining_timeout_s = calculate_remaining_timeout(
             timeout_s=timeout_s,
             start_time_s=start_time_s,
             curr_time_s=time.time(),
         )
-        return ray.get(self._obj_ref, timeout=remaining_timeout_s)
+        return ray.get(object_ref, timeout=remaining_timeout_s)
 
     @_process_response
     async def get_async(self):
         assert (
-            self._obj_ref is not None or not self._is_streaming
-        ), "get_async() can only be called on a non-streaming ActorReplicaResult"
+            not self._is_streaming
```

Comment:
Do we need these assertion statements if they are all going to be caught in `to_object_ref` and `to_object_ref_async`?